doc/reported_states_during_deployment.md

```diff
@@ -0,0 +1,353 @@
+| .properties.reported.systemModules.edgeHub.settings.image | Current image being run by container runtime. |
+| .properties.reported.modules.*\<module\>*.runtimeStatus | See [module status table below](#module-and-system-module-runtime-status). |
+| .properties.reported.modules.*\<module\>*.settings.image | Current image being run by container runtime.  The image names are read from the container itself, so will reflect the currently running image. |
+
+#### Last Desired Status codes
+
+| Code | Meaning |
+|------|---------|
+| 200  | Successful deployment |
+| 400  | ConfigFormatError - deployment was not able to be read by edgeAgent |
+| 406  | Unknown |
+| 412  | InvalidSchemaVersion - deployment schema version not compatible with edgeAgent |
+| 417  | ConfigEmptyError - no deployment has been send to IoT Edge |
+| 500  | Failed - something has gone wrong when applying a deployment |
+
+The code will be 406(Unknown) until \$edgeAgent is able to connect to the IoT Hub service, and 
+will be set to 406 when \$edgeAgent shuts down.
+
+Once online, the code may be 417(ConfigEmptyError) until \$edgeAgent receives a non-empty 
+deployment.
+
+Once the deployment is successfully realized, the code will be 200(Ok).
+
+If the deployment is not successfully realized, the code will either be 400, 412, 417 or 500, based on the
+reason why it failed.
+
+#### Module and system module runtime status
+
+A module's runtime status is the current status of the modules as interpreted by the IoT Edge 
+runtime from the container status. 
+
+| status  | Meaning |
+|---------|---------|
+| unknown | Module status is unknown. |
+| backoff | Module has had problems running, so \$edgeAgent has scheduled the module to restart at a future time.|
+| running | Module is currently running. |
+| stopped | Module has exited successfully (with a zero exit code). |
+| failed  | Module has exited with a failure exit code (non-zero). |
+
+# How these fields relate when applying a deployment
+
+The following section shows how you might detect the status of an IoT Edge Deployment, based on the fields defined above.
+
+### Before IoT Edge applies current deployment
+
+When the twin fields are in this state, the IoT Hub has applied a deployment, but the IoT Edge has
+not received and processed it.
+
+```
+configurations.<deployment name>.Status: "Applied" && 
+.properties.desired."$version" != .properties.reported.lastDesiredVerison 
+```
+ 
+
+### Completely successful
+
+A deployment may report a 200 code, which means it successfully deployed all modules, and all modules
+have come into the correct state.
+
+
+```
+configurations.<deployment name>.Status: "Applied" && 
+.properties.desired."$version" == .properties.reported.lastDesiredVerison && 
+.properties.reported.lastDesiredStatus.code == 200 && 
+.properties.desired.systemModules.edgeHub.status == .properties.reported.systemModules.edgeHub.status &&
+```
+
+For each desired module:
```

Comment:
didn't catch upon my first read that this was a continuation of the previous section. Maybe consolidating this in a single block would work better. Or replacing "For each desired module" with "and for each desired module"